rpc/src/responses/json_array.rs

```diff
@@ -0,0 +1,39 @@
+use super::ParseError;
+use serde::de;
+use serde_json::{self, json, Value};
+
+pub struct JsonArray<T> {
+    flattened_vec: Vec<T>,
+}
+
+impl<T: de::DeserializeOwned> JsonArray<T> {
+    pub fn from_str(str_to_parse: &str) -> Result<Self, ParseError> {
+        let mut parse_response = serde_json::from_str(str_to_parse).unwrap_or_else(|_| json!([]));
```

Comment:
Why do we use `unwrap_or_else` here? If `serde_json::from_str(str_to_parse)` throws an error because `str_to_parse` was gibberish, wouldn't we want to return a ParseError? I think `unwrap_item_in_nested_json_array` handles the empty case below, but why not just handle it here with a `unwrap`?

Reply:
I think @astherath was the one to add this but if he agrees I can changes this so it throws a ParseError instead of continuing with an empty array

Reply:
yep, code was written pre `Result` handling and didn't change after. Will add re-implementation.